### DISCUSSION TITLE
How forks works on Codewars

### DISCUSSION THREAD

**Discussion by Alexenon** [codewars/codewars.com#3001] [Q&A]
Hello everyone, I have a couple questions on how forks on Codewars, especially how to merge a fork. Here are my questions:

1. How does the forking mechanism work on Codewars kata?
2. What happens when I fork a kata? 
3. How do I merge a fork into a solution? 
4. What should I do with pending forks on my authored kata ?

### Example:
I uploaded a solution for a kata. After a while, I found a better algorithm, and now I want to upgrade my own solution. I fork my solution, replace with a new one, and I can't find the merging button. I also don't find where can I merge pending forks for my authored kata. I'm hoping some experienced members can shed some light on the following.

_Any insights or tips on forking katas would be greatly appreciated!_

**Comment by Blind4Basics** [comment]
There are 2 different kinds of forks.

### 1. solution forks

those are only alternative "branches" of the solution. No merging operation on them. At all.

### 2. kata forks

Those aren't available to all users (I don't remember the exact rules at the moment, but basically: must have enough honor to unloke a privilege, and must have solved the kata to see the button. Not talking about special users, like menders/mods/admins, whose the capabilities are slightly different).
* Forking the kata can be done in one language at a time, and it allows to modify the latest/current version of the kata in that language. 
* Once published, the fork must be approved by another user with the required privileges (unless blabla-mods/admins, or you're the author of the kata itself)

Forks are the proper way to fix a kata nowadays, without entering its edit panel. They are a replacement for translations (which were causing a _lot_ of problems), except when the goal is to create a new language for a kata. Then translation is the way to go.

Clearer?

**Reply by Alexenon** [reply]
Yes, but not at all 😄 

Upon your statement 

> "I don't remember the exact rules at the moment"

Will be great to add a documentation page or to add here some generic rules to users that want to use the forking mechanism to improve a kata.

Another question that comes after this phrase, 

> "Solution forks are only alternative "branches" of the solution. No merging operation on them. At all."

- Why do we need solution forks ? I thought they are supposed to improve an existing solution by just merging them, like on GitHub by forking, making changes and then merging if the forked solution is better.

**Reply by Blind4Basics** [reply]
docs: yeah, probably.

---

We do not _need_ forks. But they are useful to show variations or different ideas based on the same code or completely different approaches. Whatever the users like/are in the mood for, the point being that everything is rooted on the very same solution, and everything is visible to others. I often fork solutions of other users, to show them something, for example. 
So they do their job: they "fork". It's just that they cannot be merged.

**Comment by Molksy** [comment]
1) How does the forking mechanism work on Codewars kata?
When you fork a kata on Codewars, you create a copy of the original kata in your own personal workspace. This allows you to make changes to the kata without affecting the original version. You can then submit your solution to your forked version of the kata, and it will be graded independently of the original kata.

2) What happens when I fork a kata?
When you fork a kata, a copy of the original kata is created in your personal workspace. You can then make changes to the kata, such as adding new test cases or changing the instructions. Any changes you make will only affect your forked version of the kata and will not affect the original kata.

3) How do I merge a fork into a solution?
To merge a fork into your solution, you can follow these steps:

- [ ] Go to your forked version of the kata.
- [ ] Click on the "Pending Forks" tab.
- [ ] Find the fork you want to merge and click on "Merge".
- [ ] This will bring up a comparison of the changes between your fork and the original version of the kata.
- [ ] Review the changes and click on "Merge Changes" to merge the fork into your solution.

4)What should I do with pending forks on my authored kata?
When someone forks your kata and submits a solution, their fork will show up under the "Pending Forks" tab on your kata page. To merge their solution into your kata, you can follow the steps outlined in question 3 above.

**Reply by Blind4Basics** [reply]
wtf? half of that message is just outright false, talking about mechanisms that do not even _exist_ on CW... :rofl: 

Please talk about things you know, don't assume CW works like github just because you find the word "fork" in there.

**Reply by Alexenon** [reply]
I am quite sure that @Molksy's answer is generated by AI, because I searched it by my own to find answers for my questions. Overall, most of the answers are wrong and are just auto generated by GPT as a thought how it would work the fork system on codewars.

**Reply by Blind4Basics** [reply]
yeah, that's the feeling I got as well.

**Reply by Molksy** [reply]
> I am quite sure that @Molksy's answer is generated by AI, because I searched it by my own to find answers for my questions. Overall, most of the answers are wrong and are just auto generated by GPT as a thought how it would work the fork system on codewars.

Of course this is AI's answer. It's even easy to check if you ask him. But until I wrote anything here - no one even thought to pay attention to the author's question xD


Cunningham's Law states that "the best way to get the right answer on the Internet is not to ask a question, but to post a false answer."
Maybe someone will write the correct answer

**Reply by Blind4Basics** [reply]
do you realize you look like a troll, doing this...?

**Reply by Molksy** [reply]
> do you realize you look like a troll, doing this...?

No, sorry if it looks like that. I don’t get any benefit from this, moral satisfaction either.
If I offended or offended - I apologize. I can delete posts.

**Reply by Blind4Basics** [reply]
There is no problem, no. I'm just pointing that you shouldn't be surprised if you do this anywhere at anytime, and get kicked out/banned at some point. x)
